src/tests/Microsoft.PowerFx.Interpreter.Tests/DatabaseTests.cs

```diff
@@ -0,0 +1,212 @@
+    public class DatabaseTests
+    {
+        [Fact]
+        public void PatchFirst()
+        {
+            string expr = @"Patch(t, First(t), { Name: ""John""})";
+            (PowerFxConfig config, RecalcEngine engine, CheckResult checkResult) = CheckExpression(expr);
+            (SymbolValues values, TestDatabaseTableValue database) = GetData(config);
+
+            FormulaValue result = checkResult.GetEvaluator().Eval(values);
+            Assert.False(result is ErrorValue);
+
+            TestDatabaseRecordValue[] records = database.Records;
+            Assert.Equal(2, records.Length);
+            Assert.Equal("John", records[0].Name);
+            Assert.True(records[0].TryGetPrimaryKey_Called);
+            Assert.False(records[1].TryGetPrimaryKey_Called);
+        }
+
+        [Fact]
+        public void PatchById()
+        {
+            // The key difference here with previous test is that the record we want to delete will have a RecordValue type
+            // As a result, TryGetPrimaryKey will fail and all fields will be compared to each rows of the DB
+            string expr = @"Patch(t, {Id: 2, Name: ""Mike"", Val: ""Val2""}, { Name: ""John""})";
+            (PowerFxConfig config, RecalcEngine engine, CheckResult checkResult) = CheckExpression(expr);
+            (SymbolValues values, TestDatabaseTableValue database) = GetData(config);
+
+            FormulaValue result = checkResult.GetEvaluator().Eval(values);
+            Assert.False(result is ErrorValue);
+
+            TestDatabaseRecordValue[] records = database.Records;
+            Assert.Equal(2, records.Length);
+            Assert.Equal("John", records[1].Name);
+            Assert.False(records[0].TryGetPrimaryKey_Called);
+            Assert.False(records[1].TryGetPrimaryKey_Called);
+        }
+
+        [Fact]
+        public void RemoveFirst()
+        {
+            string expr = @"Remove(t, First(t))";
+            (PowerFxConfig config, RecalcEngine engine, CheckResult checkResult) = CheckExpression(expr);
+            (SymbolValues values, TestDatabaseTableValue database) = GetData(config);
+
+            FormulaValue result = checkResult.GetEvaluator().Eval(values);
+            Assert.False(result is ErrorValue);
+
+            TestDatabaseRecordValue[] records = database.Records;
+            Assert.Single(records);
+            Assert.Equal("Mike", records[0].Name);
+        }
+
+        private (PowerFxConfig config, RecalcEngine engine, CheckResult checkResult) CheckExpression(string expr)
+        {
+            PowerFxConfig config = new PowerFxConfig(Features.PowerFxV1);
+            RecalcEngine engine = new RecalcEngine(config);
+
+            config.EnableSetFunction();
+            config.SymbolTable.EnableMutationFunctions();
+
+            RecordType recordType = TestDatabaseRecordValue.CustomRecordType;
+            config.SymbolTable.AddVariable("t", recordType.ToTable(), new SymbolProperties() { CanSet = true, CanMutate = true });
```

Comment:
>config.SymbolTable.AddVariable( [](http://example.com/codeflow?start=12&length=31)

engine.UpdateVariable() ? may be simpler

Reply:
UpdateVariable doesn't allow providing SymbolProperties, leading to IMutationCopy requirement which makes things way more complicated

Reply:
ok - although we could add that overload.